### PR TITLE
feat(api): (2) Upgrade upload APIs to Version 2

### PR DESCRIPTION
**Pull request by Akashsah2003**
## Description

This PR aims to do tasks 7, 8, 12, 14, 15, 16, 17 of issue #2623.

### Changes

1. **Moved `groupName` variable from headers to query parameters**

2. **/updoads**, POST
- Moved `folderId` , `uploadDescription`, `public`, `applyGlobal`, `ignoreScm`, `uploadType` variables from the headers to body parameters.
- Changed unit test for the controller.
3. **/uploads/{id}/agents**, GET
- Changed these fields to camelCase: `agent_id`, `agent_rev`, `agent_name`
4. **/uploads/{id}/licenses/edited**, GET
- Changed these fields to camelCase: `spdx_id`
5. **/uploads/{id}/licenses/scanned**, GET
- Changed these fields to camelCase: `shortname`
6. **/uploads/{id}/item/{itemId}/tree/view**, GET
- Moved `page` , `limit` variables from the headers to query parameters.

## How to test

- Changes pass all the unit tests.
- Can be tested manually using postman or swagger.

/cc @dushimsam @GMishx @shaheemazmalmmd

### CODE REVIEW

**Comment by dushimsam** [comment]
Hello @Akashsah2003 , these changes should go a long with updating the API documentation of the version-2 APIs

**Comment by Akashsah2003** [comment]
Hello @dushimsam, changes made to documentation. Please take a look.

File: src/www/ui/api/Controllers/UploadController.php [hunk 1]
```diff
@@ -417,11 +417,11 @@ private function changeUpload($request, $response, $args, $isCopy)
    */
   public function postUpload($request, $response, $args)
   {
-    $uploadType = $request->getHeaderLine('uploadType');
+    $reqBody = $this->getParsedBody($request);
+    $uploadType = $reqBody['uploadType'] ?? null;
```
Comment (GMishx) [inline src/www/ui/api/Controllers/UploadController.php:421]: The API should still support V1 requests in deprecated mode until it is completely removed. You can read the version of request with `ApiVersion::getVersion($request) == ApiVersion::V2`.

See https://github.com/fossology/fossology/pull/2617/files#diff-1bae1ab555e908fff79a5b02ff0f05677a8d1f5653259e98531071acfeb7e16bR42 for reference.

File: src/www/ui/api/Controllers/UploadController.php [hunk 2]
```diff
@@ -1093,7 +1093,13 @@ public function getAllAgents($request, $response, $args)
         $item['currentAgentRev'] = "";
       }
       foreach ($item['successfulAgents'] as &$agent) {
-        $agent['agent_id'] = intval($agent['agent_id']);
+        $agent['agentId'] = intval($agent['agent_id']);
+        $agent['agentRev'] = $agent['agent_rev'];
+        $agent['agentName'] = $agent['agent_name'];
+        unset($agent['agent_id']);
+        unset($agent['agent_name']);
+        unset($agent['agent_rev']);
```
Comment (GMishx) [inline src/www/ui/api/Controllers/UploadController.php:1101]: I would instead recommend to create a new model and get the version aware array from it. See https://github.com/fossology/fossology/pull/2617/files#diff-f62f1d2b860c504c87d386cdfa79513c109848593f0afeb60a5dc76e4b98ff42 for reference.

File: src/www/ui/api/Middlewares/RestAuthMiddleware.php [hunk 3]
```diff
@@ -79,7 +79,8 @@ public function __invoke(Request $request, RequestHandler $handler) : ResponseIn
          */
         throw new HttpForbiddenException("Do not have required scope.");
       }
-      $groupName = $request->getHeaderLine('groupName');
+      $query = $request->getQueryParams();
+      $groupName = $query["groupName"] ?? null;
```
Comment (GMishx) [inline src/www/ui/api/Middlewares/RestAuthMiddleware.php:83]: Please keep it version aware.

File: src/www/ui/api/Controllers/UploadController.php [hunk 4]
```diff
@@ -417,11 +417,11 @@ private function changeUpload($request, $response, $args, $isCopy)
    */
   public function postUpload($request, $response, $args)
   {
-    $uploadType = $request->getHeaderLine('uploadType');
+    $reqBody = $this->getParsedBody($request);
+    $uploadType = $reqBody['uploadType'] ?? null;
     if (empty($uploadType)) {
       $uploadType = 'vcs';
```
Comment (GMishx) [inline src/www/ui/api/Controllers/UploadController.php:423]: As per the OpenAPI doc, `uploadType` is a required field and thus, should throw appropriate exception if empty.

**Review by GMishx — CHANGES_REQUESTED** [review]
Please support both version 1 and 2 of the API.

**Comment by Akashsah2003** [comment]
I have made the following changes @GMishx

- Added support for both Version1 and Version2 for apis.
- `uploadType` is now throwing bad request exception.
- Created models `Agent.php`, `SuccessfulAgent.php`, `EditedLicense.php` in order to support both API V1 and V2 responses.
- Modified Unit tests accordingly.

**Comment by Akashsah2003** [comment]
@GMishx I have made the commit so that tests pass. Kindly review the pr.

File: src/www/ui/api/Controllers/UploadController.php [hunk 5]
```diff
@@ -400,7 +404,7 @@ private function changeUpload($request, $response, $args, $isCopy)
     if (!$request->hasHeader('folderId') ||
         !is_numeric($newFolderID = $request->getHeaderLine('folderId'))) {
       throw new HttpBadRequestException("folderId header should be an integer!");
-    }
+    };
```
Comment (GMishx) [inline src/www/ui/api/Controllers/UploadController.php:407]: Irrelevant changes, please revert.

File: src/www/ui/api/Controllers/UploadController.php [hunk 6]
```diff
@@ -455,17 +478,10 @@ public function postUpload($request, $response, $args)
...
-    $public = empty($public) ? 'protected' : $public;
-    $applyGlobal = filter_var($request->getHeaderLine('applyGlobal'),
-      FILTER_VALIDATE_BOOLEAN);
-    $ignoreScm = $request->getHeaderLine('ignoreScm');
-
     $locationObject = [];
     if (array_key_exists("location", $reqBody)) {
       $locationObject = $reqBody["location"];
-    } elseif ($request->getHeaderLine('uploadType') != 'file') {
+    } elseif ($reqBody['uploadType'] != 'file') {
```
Comment (GMishx) [inline src/www/ui/api/Controllers/UploadController.php:484]: Better use `$uploadType` variable already assigned based on API version.

File: src/www/ui/api/Models/Agent.php [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+        ];
+    }
+    return [
+        'successfulAgents' => $this->getSuccessfulAgents(),
+        'uploadId' => $this->getUploadId(),
+        'agentName' => $this->getAgentName(),
+        'currentAgentId' => $this->getCurrentAgentId(),
+        'currentAgentRev' => $this->getCurrentAgentRev(),
+        'isAgentRunning' => $this->getIsAgentRunning()
+    ];
```
Comment (GMishx) [inline src/www/ui/api/Models/Agent.php:144]: Please use 2 space indentation.

File: src/www/ui/api/Models/EditedLicense.php [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+            'count' => $this->getCount(),
+            'spdxId' => $this->getSpdxId()
+        ];
+    }
+    return [
+        'id' => $this->getId(),
+        'shortName' => $this->getShortName(),
+        'count' => $this->getCount(),
+        'spdx_id' => $this->getSpdxId()
+    ];
```
Comment (GMishx) [inline src/www/ui/api/Models/EditedLicense.php:111]: Please use 2 space indentation.

File: src/www/ui/api/Models/Agent.php (hunk 7 above)
Comment (GMishx) [reply src/www/ui/api/Models/Agent.php:144]: Both the responses (v1 and v2) looks same to me. Can the check be simply removed?

File: src/www/ui/api/Models/SuccessfulAgent.php [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+            'agentId' => $this->getAgentId(),
+            'agentRev' => $this->getAgentRev(),
+            'agentName' => $this->getAgentName()
+        ];
+    }
+    return [
+        'agent_id' => $this->getAgentId(),
+        'agent_rev' => $this->getAgentRev(),
+        'agent_name' => $this->getAgentName()
+    ];
```
Comment (GMishx) [inline src/www/ui/api/Models/SuccessfulAgent.php:96]: Please use 2 space indentation.

**Review by GMishx — CHANGES_REQUESTED** [review]
Minor changes, rest looks good.

**Comment by Akashsah2003** [comment]
Hi @GMishx I have made the required changes. Please take a look.

**Comment by Akashsah2003** [comment]
Hi @GMishx , Friendly reminder to review the latest changes in the PR. Your input is appreciated. Thanks!

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Review by GMishx — APPROVED** [review]
Tested, working as expected.

**Pull request merged**